### PR TITLE
Fix deprecations

### PR DESCRIPTION
**Pull request by txtsd**
I've replaced most of the deprecated methods with recommended ones.

I haven't touched any QTime methods since QElapsedTimer is monotonic and doesn't allow for conversion to human readable time. I've seen at least one instance of conversion to human readable time.
I also haven't touched QJSEngine because there is no alternative for the deprecated method.
There is another deprecation warning, but it didn't make sense to me so I left it alone. It's just a `return nullptr` but the output is this:
```
/home/txtsd/git/imgbrd-grabber/src/gui/src/ui/fixed-size-grid-layout.cpp: In member function ‘virtual Qt::Orientations FixedSizeGridLayout::expandingDirections() const’:
/home/txtsd/git/imgbrd-grabber/src/gui/src/ui/fixed-size-grid-layout.cpp:89:9: warning: ‘constexpr QFlags<T>::QFlags(QFlags<T>::Zero) [with Enum = Qt::Orientation; QFlags<T>::Zero = int QFlags<Qt::Orientation>::Private::*]’ is deprecated: Use default constructor instead [-Wdeprecated-declarations]
   89 |  return nullptr;
      |         ^~~~~~~
In file included from /usr/include/qt/QtCore/qglobal.h:1304,
                 from /usr/include/qt/QtGui/qtguiglobal.h:43,
                 from /usr/include/qt/QtWidgets/qtwidgetsglobal.h:43,
                 from /usr/include/qt/QtWidgets/qlayout.h:43,
                 from /usr/include/qt/QtWidgets/QLayout:1,
                 from /home/txtsd/git/imgbrd-grabber/src/gui/src/ui/fixed-size-grid-layout.h:4,
                 from /home/txtsd/git/imgbrd-grabber/src/gui/src/ui/fixed-size-grid-layout.cpp:1:
/usr/include/qt/QtCore/qflags.h:123:80: note: declared here
  123 |     QT_DEPRECATED_X("Use default constructor instead") Q_DECL_CONSTEXPR inline QFlags(Zero) noexcept : i(0) {}
      |       
```

Let me know if any behavior has changed.

### CODE REVIEW

File: src/gui/src/viewer/zoom-window.cpp [hunk 1]
```diff
@@ -432,7 +433,7 @@ void ZoomWindow::display(const QPixmap &pix, int size)
 		updateWindowTitle();
 
 		if (m_isFullscreen && m_fullScreen != nullptr && m_fullScreen->isVisible()) {
-			m_fullScreen->setImage(m_displayImage.scaled(QApplication::desktop()->screenGeometry().size(), Qt::KeepAspectRatio, Qt::SmoothTransformation));
+			m_fullScreen->setImage(m_displayImage.scaled(QGuiApplication::primaryScreen()->geometry().width(), QGuiApplication::primaryScreen()->geometry().height(), Qt::KeepAspectRatio, Qt::SmoothTransformation));
```
Comment (Bionus) [inline src/gui/src/viewer/zoom-window.cpp:436]: You could have just used `geometry().size()` to save some duplication on the call here.

File: src/gui/src/viewer/zoom-window.cpp [hunk 2]
```diff
@@ -730,7 +731,7 @@ void ZoomWindow::draw()
 		update();
 
 		if (m_isFullscreen && m_fullScreen != nullptr && m_fullScreen->isVisible()) {
-			m_fullScreen->setImage(m_displayImage.scaled(QApplication::desktop()->screenGeometry().size(), Qt::KeepAspectRatio, Qt::SmoothTransformation));
+			m_fullScreen->setImage(m_displayImage.scaled(QGuiApplication::primaryScreen()->geometry().width(), QGuiApplication::primaryScreen()->geometry().height(), Qt::KeepAspectRatio, Qt::SmoothTransformation));
```
Comment (Bionus) [inline src/gui/src/viewer/zoom-window.cpp:734]: Same thing

File: src/gui/src/viewer/zoom-window.cpp [hunk 3]
```diff
@@ -993,7 +994,7 @@ void ZoomWindow::fullScreen()
 		if (!m_isAnimated.isEmpty()) {
 			m_fullScreen->setMovie(m_displayMovie);
 		} else {
-			m_fullScreen->setImage(m_displayImage.scaled(QApplication::desktop()->screenGeometry().size(), Qt::KeepAspectRatio, Qt::SmoothTransformation));
+			m_fullScreen->setImage(m_displayImage.scaled(QGuiApplication::primaryScreen()->geometry().width(), QGuiApplication::primaryScreen()->geometry().height(), Qt::KeepAspectRatio, Qt::SmoothTransformation));
```
Comment (Bionus) [inline src/gui/src/viewer/zoom-window.cpp:997]: Same thing

**Review by Bionus — APPROVED** [review]
LGTM, thanks!

I'll see how much it prevents earlier systems from building it. Even though I don't mind dropping support for building on old systems.

File: src/gui/src/viewer/zoom-window.cpp (hunk 1 above)
Comment (txtsd) [reply src/gui/src/viewer/zoom-window.cpp]: Replaced with size()

File: src/gui/src/viewer/zoom-window.cpp (hunk 2 above)
Comment (txtsd) [reply src/gui/src/viewer/zoom-window.cpp]: Replaced with size()

File: src/gui/src/viewer/zoom-window.cpp (hunk 3 above)
Comment (txtsd) [reply src/gui/src/viewer/zoom-window.cpp]: Replaced with size()

**Review by Bionus — APPROVED** [review]

**Pull request merged**